src/lib/y2storage/devicegraph.rb

```diff
@@ -559,5 +557,79 @@ def udev_lookup_possible?
       # called provides a seasonable result.
       !StorageManager.instance.committed?
     end
+
+    # Encryption devices might be probed with a name that does not match the device name
+    # indicated in the fstab file. For example, /etc/fstab could have an entry like:
+    #
+    #   /dev/mapper/cr_home   /home   ext4  defaults  0   0
+    #
+    # But that encryption device could be probed with a name like /dev/mapper/cr-auto-1. In that
+    # case, the device could not be found in the devicegraph when searching for the device name in the
+    # fstab entry. But, if the crypttab file was previously parsed (see Encryption#save_crypttab_names),
+    # the Encryption devices are populated in the devicegraph with their corresponding name indicated
+    # in the crypttab. This information can be used to try possible alternative names for the encryption
+    # device. For example, when the devicegraph contains a Encryption layer /dev/mapper/cr-auto-1 over
+    # the device /dev/sda1, and the /etc/crypttab has the following entry:
+    #
+    #   cr_home   /dev/sda1
+    #
+    # a possible alternative name for /dev/mapper/cr_home would be /dev/mapper/cr-auto-1, due to the
+    # encryption device cr-auto-1 has "cr_home" as crypttab_name (after parsing the crypttab file).
+    #
+    # @param device_name [String] a kernel name or udev name
+    # @return [Array<String>]
+    def alternative_names(device_name)
+      devices = encryptions.select { |e| e.crypttab_name? && device_name.match?(e.crypttab_name) }
```

Comment:
Why is `#match?` used here to compare the strings?

Reply:
Here strings are not exactly compared. It is checked that crypttab_name is included in the device_name, for example:

```
device_name #=> "/dev/mapper/cr_home"
device.crypttab_name #=> "cr_home"

device_name.match?(device.crypttab_name) #=> true
```

BTW, maybe `String#include?` is a better method for this.

Reply:
For this case I would use `String#include?`

Reply:
Yes, with `include?` the intention is much more clear.